src/H5Lpublic.h

```diff
@@ -321,8 +322,8 @@ H5_DLL herr_t H5Lcreate_hard_async(hid_t cur_loc_id, const char *cur_name, hid_t
  *
  *          \p link_loc_id and \p link_name specify the location and name,
  *          respectively, of the new soft link. \p link_name is interpreted
- *          relative to \p link_loc_id and must contain only the name of the soft
- *          link; \p link_name may not contain any additional path elements.
+ *          as a path relative to \p link_loc_id, or an absolute path if it
+ *          begins with \c /.
```

Comment:
Maybe say something about how the actual name of the link created is only the last element of the path, and prior elements are used to find the group in which to create the link.